test/views/supplementary/PreviousDocumentsViewSpec.scala

```diff
@@ -0,0 +1,224 @@
+      assertMessage(hint, "For example, 8GB-123456789101-SHIP1")
+      assertMessage(documentX, "Temporary storage")
+      assertMessage(documentY, "Simplified declaration")
+      assertMessage(documentZ, "Previous document")
+      assertMessage(documentType, "Enter previous document code")
+      assertMessage(documentReference, "Enter the reference for the previous DUCR or MUCR")
+      assertMessage(documentGoodsIdentifier, "Enter the goods item this previous document relates to")
+      assertMessage(documentCategoryLabel, "Document Category")
+      assertMessage(documentTypeLabel, "Document Type")
+      assertMessage(documentReferenceLabel, "Document Reference")
+      assertMessage(documentGoodsIdentifierLabel, "Goods Item Identifier")
+      assertMessage(removePackageInformation, "Remove Packaging Information")
+    }
+
+    "have proper messages for error labels" in {
+
+      assertMessage(documentCategoryEmpty, "Document category cannot be empty")
+      assertMessage(documentCategoryError, "Document category is incorrect")
+      assertMessage(documentTypeEmpty, "Previous document code cannot be empty")
+      assertMessage(documentTypeError, "Previous document code is incorrect")
+      assertMessage(documentReferenceEmpty, "Reference for the DUCR or MUCR cannot be empty")
+      assertMessage(documentReferenceError, "Reference for the DUCR or MUCR is incorrect")
+      assertMessage(documentGoodsIdentifierError, "This field is incorrect")
+    }
+  }
+
+  "Previous Documents View on empty page" should {
+
+    "display page title" in {
+
+      getElementByCss(createView(), "title").text() must be(messages(previousDocuments))
+    }
+
+    "display header with hint" in {
+
+      val view = createView()
+
+      getElementByCss(view, "legend>h1").text() must be(messages(title))
+      getElementByCss(view, "legend>span").text() must be(messages(hint))
+    }
+
+    "display three radio buttons with description (not selected)" in {
+
+      val view = createView(Document.form().fill(Document("", "", "", Some(""))))
+
+      val optionOne = getElementById(view, "Temporary storage")
+      optionOne.attr("checked") must be("")
+
+      getElementByCss(view, "#documentCategory>div:nth-child(2)>label").text() must be(messages(documentX))
+
+      val optionTwo = getElementById(view, "Simplified declaration")
+      optionTwo.attr("checked") must be("")
+
+      getElementByCss(view, "#documentCategory>div:nth-child(3)>label").text() must be(messages(documentY))
+
+      val optionThree = getElementById(view, "Previous document")
+      optionThree.attr("checked") must be("")
+
+      getElementByCss(view, "#documentCategory>div:nth-child(4)>label").text() must be(messages(documentZ))
+    }
+
+    "display empty input with label for Previous document code" in {
+
+      val view = createView()
+
+      getElementByCss(view, "form>div:nth-child(4)>label>span").text() must be(messages(documentType))
```

Comment:
We meet again...

Reply:
As we spoke before, we just can't do it without proper ids naming - I can try finding it differently but it still could be a bit messt (xpath, name etc)